packages/js/components/src/tag/index.tsx

```diff
@@ -28,72 +32,84 @@ type Props = {
 	className?: string;
 };
 
-const Tag: React.VFC< Props > = ( {
-	id,
-	instanceId,
-	label,
-	popoverContents,
-	remove,
-	screenReaderLabel,
-	className,
-} ) => {
-	const [ isVisible, setIsVisible ] = useState( false );
+const Tag = forwardRef(
+	(
+		{
+			id,
+			label,
+			popoverContents,
+			remove,
+			screenReaderLabel,
+			className,
+		}: Props,
+		removeButtonRef: Ref< HTMLButtonElement >
+	) => {
+		const [ isVisible, setIsVisible ] = useState( false );
 
-	screenReaderLabel = screenReaderLabel || label;
-	if ( ! label ) {
-		// A null label probably means something went wrong
-		// @todo Maybe this should be a loading indicator?
-		return null;
-	}
-	label = decodeEntities( label );
-	const classes = classnames( 'woocommerce-tag', className, {
-		'has-remove': !! remove,
-	} );
-	const labelId = `woocommerce-tag__label-${ instanceId }`;
-	const labelTextNode = (
-		<Fragment>
-			<span className="screen-reader-text">{ screenReaderLabel }</span>
-			<span aria-hidden="true">{ label }</span>
-		</Fragment>
-	);
+		const instanceId = useInstanceId( Tag ) as string;
```

Comment:
Switched from a HOC to a hook since I wasn't able to make it work alongside `forwardRef`